typeguard/__init__.py

```diff
@@ -595,6 +608,9 @@ def check_type(argname: str, value, expected_type, memo: Optional[_CallMemo] = N
     elif isinstance(expected_type, TypeVar):
         # Only happens on < 3.6
         check_typevar(argname, value, expected_type, memo)
+    elif BPLiteral is not None and isinstance(expected_type, BPLiteral.__class__):
+        # Only happens on < 3.7 when using Literal from typing_extensions
```

Comment:
Presumably happens on any Python version when using Literal from typing_extensions, no?

Reply:
`typing_extensions` is smart and will return the standard `Literal` implementation if it exists.

```python3
# Python 3.8.0 (default, Oct 28 2019, 16:14:01)
# [GCC 8.3.0] on linux
# Type "help", "copyright", "credits" or "license" for more information.
>>> from typing import Literal
>>> from typing_extensions import Literal as BPLiteral
>>> assert Literal is BPLiteral
>>>
```

In this case, the origin type checker dict lookup will yield `check_literal` and this code path will not be taken.

Reply:
On Python 3.7, the implementation of `Literal` is different in `typing_extensions` than in Python 3.6. Namely, the generics typing API is used, so in this case the origin type checker dict lookup will again yield `check_literal`. This is why I've added `check_literal` to the dict under `BPLiteral` as well.

Reply:
Ok, got it.